### PR TITLE
Semi-randomized beer kegs on the Cargo supply shuttle.

### PR DESCRIPTION
**Pull request by Eneocho**
Cargo can now purchase Beer kegs from small town breweries for 60 credits. These beer kegs can contain a varied set of chems that can range from juices to assorted drinks or sometimes even medical compounds.
Hacking the console, however, lets you purchase off-brand beer kegs, for 40 credits. These contain dangerous compounds sometimes, as they are the failed results of beer research, but also have a higher chance for special compounds.
Being on the DOES NOT PAY TIP blacklist (from the syndicate tool) makes it so the kegs (either) are very contaminated with assorted toxic, dangerous or otherwise non-edible compounds.
The names and descriptions are semi-random, pulling from a list, so you can get stuff like these two examples:
image
image
In place of custom randomized icons, the beer kegs are randomly colored upon spawn.
## Why it's good
You can now party with extreme class by drinking the designer beer.
## How it was tested
Spawned these beer kegs, worked fine. Bought them, worked fine.
Used the DOES NOT PAY TIP backdoor then spawned and bought some, worked as intended.

Ideas for more names or chems are welcome, and suggestions if some chems available here are too strong or not.
:cl:
 * rscadd: You can now buy semi-randomized small town brewery beer kegs in cargo for 60 credits, these are safe.
 * rscadd: Hacking the cargo console lets you buy the off-brand beer kegs for 40 credits, which can contain dangerous chemicals, but also extra special chems.
 * rscadd: These beer kegs will be VERY contaminated (at least 1 dangerous chemical on average) if the station is on the DOES NOT TIP backdoor.
 * bugfix: Adds Iced Beer to the COLDDRINKS define (i had forgotten iced beer was a thing).
 * tweak: The Does Not Tip backdoor uplink description also mentions it poisoning beers.

### CODE REVIEW

**Comment by angryturnip** [comment]
Should make diabeetusol one of the possible bad chems to give a possible but unlikely way to get it without an emag as a traitor if for some reason you don't want to buy an emag.

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -392,10 +440,6 @@
 /obj/structure/reagent_dispensers/bloodkeg/cultify()
 	return
 
-/obj/structure/reagent_dispensers/beerkeg/blob_act()
-	explosion(src.loc,0,3,5,7,10)
```
Comment (west3436) [inline code/modules/reagents/reagent_dispenser.dm:396]: You WHAT??

Comment (Eneocho) [reply code/modules/reagents/reagent_dispenser.dm:396]: I only moved that blob act, not removed it

**Comment by djkramer123** [comment]
Seems like something that should be in the soldering iron hacked stock rather than the "official" regular stock if it isn't already

**Comment by Eneocho** [comment]
Yeah I was kinda thinking about that, still on the fence about how it would be best implemented.
Was thinking of:
* Fully safe on regular stock
* Slight odds of toxic shit on contraband stock
* Balls to the walls extremely toxic if on the no tip backdoor

Would make it so it's also extremely toxic if emagged, but not sure how to make it so a supply datum is hidden by emagging (as to not have THREE whole variant), or make it so the kegs themselves check if the console was emagged, then change to be contaminated ones.

File: code/modules/reagents/reagent_dispenser.dm (hunk 1 above)
Comment (west3436) [reply code/modules/reagents/reagent_dispenser.dm:396]: I was more surprised that they even explode at all but if you're just maintaining existing behavior then right on

**Comment by west3436** [comment]
Any reason why you didn't just make this a subtype of the existing beer keg?

**Comment by Eneocho** [comment]
> Any reason why you didn't just make this a subtype of the existing beer keg? 

It would be filled with 1000u beer and postponing the ..() call to the end would make it so it doesn't initialize the 1000u container.
This is probably the same reason as to why the blood keg is it's own thing instead of a child of the beer keg.
Do not merge this yet, please, I have some small changes to make still, but I'm busy right now.

**Comment by Eneocho** [comment]
ok it's merge ready

**Comment by west3436** [comment]
> ok it's merge ready

No it's not, you haven't made sprites yet

**Comment by Eneocho** [comment]
> No it's not, you haven't made sprites yet

I haven't, but I rigged it such that the kegs come with random colors selected from #4D4D4D to #FFFFFF, which does the job for now. I can't sprite for shit right now.

**Comment by Eneocho** [comment]
> > No it's not, you haven't made sprites yet
> 
> I haven't, but I rigged it such that the kegs come with random colors selected from #4D4D4D to #FFFFFF, which does the job for now. I can't sprite for shit right now.

**Comment by Eneocho** [comment]
workin on some stuff

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -375,6 +375,153 @@
...
+					basechem = null
+					if(prob(50))
+						src.reagents.add_reagent(CLONEXADONE, 166)
+						src.reagents.add_reagent(BLOOD, 830)
+						amount_per_transfer_from_this = 6 //exactly one meat slab per cycle
+					else
+						src.reagents.add_reagent(ENZYME, 110)
+						src.reagents.add_reagent(MILK, 880)
+						amount_per_transfer_from_this = 45 //you know the drill by now
+					if(prob(50))
```
Comment (jwhitak) [inline code/modules/reagents/reagent_dispenser.dm]: Move this NOREACT block above the adding reagents procs, lol

File: code/modules/reagents/reagent_dispenser.dm [hunk 3]
```diff
@@ -375,6 +375,153 @@
...
+	var/selectedsecondname = pickweight(secondname) //only the clownkeg has a chance to not have a second name, so we call for one here just in case
+
+	var/list/safeingredients = list(APPLEJUICE, BANANA, LEMONJUICE, PLUMPHJUICE, WATERMELONJUICE, SUGAR, CORNSYRUP, MINTESSENCE, GARGLEBLASTER, WHISKEY, VODKA, TRIPLESEC, ICE)
+	var/list/specialingredients = list()
+	var/list/dangerousingredients = list(DANS_WHISKEY, POISONBERRYJUICE, BLISTEROL, CHEMICAL_WASTE, VIRUSFOOD, IMPEDREZENE, SALTWATER, SIMPOLINOL, SUX, PWINE, MUTAGEN, PLASMA, VOMIT, CYANIDE, DIAMONDDUST, MERCURY, FROSTOIL, CONDENSEDCAPSAICIN, SILICATE, NEUROTOXIN, DIABEETUSOL, LUBE, MOMMIMILK, PACID, PUNCTUALITE)
+
+	switch(rand(1,520))
+		if(1 to 120)
+			//normal shit
+			selectedfirstname = pickweight(firstname)
```
Comment (jwhitak) [inline code/modules/reagents/reagent_dispenser.dm]: This line can be removed because you've already put a safeguard for it above

**Review by jwhitak — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_dispenser.dm (hunk 2 above)
Comment (Eneocho) [reply code/modules/reagents/reagent_dispenser.dm]: It worked fine on testing (i think) but sure. Fixing it when I get home.

File: code/modules/reagents/reagent_dispenser.dm (hunk 3 above)
Comment (Eneocho) [reply code/modules/reagents/reagent_dispenser.dm]: fixed

**Review by jwhitak — APPROVED** [review]

**Comment by Eneocho** [comment]
Found a typo.
Reminder to check and fix that when I get home.

**Comment by west3436** [comment]
> Found a typo.
> Reminder to check and fix that when I get home. 

Repo banned.

**Pull request merged**